### ISSUE TITLE
Modal windows theme does not follow the general theme of the application

### ISSUE THREAD

**Issue by mmahmoudian** [rstudio/rstudio#10296]
### System details

    RStudio Edition : Desktop and Server
    RStudio Version : RStudio 2021.09.1+372 "Ghost Orchid" Release (8b9ced188245155642d024aa3630363df611088a, 2021-11-08) for Ubuntu Bionic
                      Mozilla/5.0 (X11; Linux x86_64; rv:95.0) Gecko/20100101 Firefox/95.0
    OS Version      :   Manjaro  Linux (5.10.89-1-MANJARO x86_64)
    R Version       :  4.1.2 (To my understanding, this is irrelevant)

### Steps to reproduce the problem

1. Open Rstudio
2. Set the theme to something other than default (e.g "Idle Fingers" which is my favorite)
3. Apply the changes (close and reopen Rstudio if needed)
4. Open any modal window (e.g "menu > Tools > Edit Code Snippets...")

### Describe the problem in detail

The issue is that the modal windows does not follow the theme that is set by user. This is especially hurtful when user has a dark theme set and they have to open some large modal window like "menu > Tools > Modify keyboard Shortcuts...".

This is consistent with all the modal windows including but but necessarily limited to:
- About Rstudio (menu > Help > About Rstudio
- Everything you find in Tools menu (except those with a keyboard shortcut):
    ![image](https://user-images.githubusercontent.com/390889/147926542-4b70b498-f5b3-4217-b2da-af7a38c83905.png)
- Open Profile (menu > Profile > Open Profile...)
- Open file
- Save As
- Import Dataset (menu > File > Import Dataset > *)

For more examples and illustrations, visit #10277.

Interesting enough, the "menu > File > Print..." follows the user's theme!

### Describe the behavior you expected

I expect the modal windows to follow the same theming style the user has chosen. At least it respects the dark or light general schema of the chosen theme.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@mmahmoudian Thank you for raising the issue! 
1) I can't quite reproduce the issue—even the color doesn't get applied.
2) This might be a different issue, but changing the system's theme from light to dark isn't getting recognized.
3) I thought there were already issues for this, but I can't find any.

Note: I don't think this is new, as it appears to be an issue in 1.4.1106, as well.

**Comment by mmahmoudian** [comment]
> I can't quite reproduce the issue

I don't understand! Are you saying that the modal windows on your computer are following the theme change? I've been using Rstudio since 2013 practically every day and all versions and have never ever seen them getting the theme that user chooses. Is there a setting that I've missed?

**Comment by ronblum** [comment]
@mmahmoudian Sorry about that—I worded that very badly. I thought that in your screenshot, the dropdown menu was getting changed by the theme, not just the modal. But I imagine that it's dark because your system is in dark mode?

**Comment by mmahmoudian** [comment]
Yes, the menu correctly followed the theme (dark in my case), but if you click on anything in that menu it opens a modal with white theme. The reason I added the screenshot of the Tools menu was because I was too lazy to type all the items in that list 😅

I hope now the issue I'm raising is more clear. If there are still questions/doubts, I can record some short video and explain.

**Comment by gtritchie** [comment]
This has existed since we first added editor theming support back in 1.1 (I think it was in 1.1).

There are some fundamentally different types of modal dialogs in the IDE, and the behavior can vary between them.

- Native desktop system modal dialogs; these are things like File Open, which are brought up in the Desktop IDE to ensure we get full desktop functionality for browsing the local filesystem; these may obey the system dark mode (will vary by platform) but will always be unaware of the RStudio Editor Theme setting
- Native modal dialogs implemented via Qt (also desktop-only); there aren't too many of these, but they will also ignore the IDE Editor Theme (but may obey the system dark mode)
- Web-based modal dialogs; you can tell which ones they are because if you move them around they can't leave the bounds of the IDE window since they are just part of the web page that makes up the IDE (both desktop and server); these do not obey theming of the system dark mode OR the RStudio IDE theme. MOST of the "dialogs" in the RStudio IDE are of this type.

I also thought there was an existing issue on this, but couldn't find it.

**Comment by mmahmoudian** [comment]
@gtritchie You are correct, and also correct that majority of modals are "Web-based modal dialogs" and those are the one that I've wrote this issue for. I'm on KDE and I have sark theme on everything (Qt, GTK, KDE, you name it) and the only thing that stands out hurtfully is Rstudio's Web-based modals.

**Comment by mmahmoudian** [comment]
Bump to remove stale label

**Comment by mmahmoudian** [comment]
Bump to remove stale label

**Comment by mmahmoudian** [comment]
Bump to remove stale

**Comment by mmahmoudian** [comment]
Bump for removing the stale label.

Imho stale bot is most useless and irritating thing one can do to their repo. Completely counter productive.

When something has been labelled as "bug" by moderators, the stale bot should not screw it up and close the issue.

**Comment by mmahmoudian** [comment]
Bump for removing the stale label.

**Comment by mmahmoudian** [comment]
Bump for removing the stale label.

**Comment by mmahmoudian** [comment]
Bump for removing the stale label.

**Comment by ronblum** [comment]
Verified in RStudio Desktop 2026.04.0-daily+471 on Windows 11.
